### PR TITLE
Support of CORS for map services

### PR DESCRIPTION
**Pull request by laurentj**
These adds support of headers `Access-Control-Allow-*` for requests coming from external web sites.
List of authorized domain to do requests should be list into a new parameter `accessControlAllowOrigin` into `lizmapConfig.ini.php` in each repository section. 

A field to edit this parameter has been added into the form to edit parameters of a repository.

Funded by Valabre

### CODE REVIEW

File: lizmap/modules/lizmap/controllers/service.classic.php [hunk 1]
```diff
@@ -807,6 +858,7 @@ protected function GetFeature($wfsRequest)
         if (substr($result->data, 0, 7) == 'file://' && is_file(substr($result->data, 7))) {
             $rep->fileName = substr($result->data, 7);
             $rep->deleteFileAfterSending = true;
+            $rep->content = null;
```
Comment (rldhont) [inline lizmap/modules/lizmap/controllers/service.classic.php:861]: PHPStan:
```
Property jResponseBinary::$content (string) does not accept null. 
```

Comment (laurentj) [reply lizmap/modules/lizmap/controllers/service.classic.php:861]: the code is good. This is the comment of the `content` property which is not complete. I fixed it.

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 2]
```diff
@@ -136,6 +136,8 @@ form.admin_section.repository.label=Id
 form.admin_section.repository.label.label=Label
 form.admin_section.repository.path.label=Local folder path
 form.admin_section.repository.allowUserDefinedThemes.label=Allow themes/javascript codes for this repository
+form.admin_section.repository.accessControlAllowOrigin.label=List of websites allowed to access to map services (WMS/WMTS/WFS...)
+form.admin_section.repository.accessControlAllowOrigin.help=List of url separated by a comma, like 'https://domain1.com,https://www.domain2.com'. Path and query parameters will be removed from the url.
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:140]:
```suggestion
form.admin_section.repository.accessControlAllowOrigin.label=List of websites allowed to access the map services (WMS/WMTS/WFS...)
form.admin_section.repository.accessControlAllowOrigin.help=List of URL separated by a comma, like 'https://domain1.com,https://www.domain2.com'. Path and query parameters will be removed from the URL.
```

File: lizmap/modules/admin/locales/en_US/admin.UTF-8.properties [hunk 3]
```diff
@@ -148,6 +150,7 @@ form.admin_section.message.data.removed.failed=The repository has not been remov
 form.admin_section.message.repository.wrong=The repository's id given does not exist.
 form.admin_section.message.path.wrong=The path given for the repository does not exist on the server.
 form.admin_section.message.path.not_authorized=The path given for the repository is not authorized.
+form.admin_section.message.accessControlAllowOrigin.bad.domain=Content is not a url or a list of url.
```
Comment (mdouchin) [inline lizmap/modules/admin/locales/en_US/admin.UTF-8.properties:153]:
```suggestion
form.admin_section.message.accessControlAllowOrigin.bad.domain=Content is not a valid URL or a list of URL.
```

**Review by mdouchin — CHANGES_REQUESTED** [review]
Thanks. Minor string changes asked, otherwise, seems ok

**Review by mdouchin — APPROVED** [review]

**Pull request merged**